source/elements/oneDAL/source/glossary.rst

```diff
@@ -0,0 +1,117 @@
+.. _glossary:
+
+=========
+Glossary
+=========
+
+Machine learning terms
+======================
+
+.. glossary::
+    :sorted:
+
+    Attribute
+        A particular property or quality of a real object or an event.
+        Has a defined type and domain.
+
+    Classification
+        A supervised machine-learning problem to assign categorical :term:`labels<Label>`
+        to :term:`feature vectors<Feature vector>` describing input objects.
+
+    Dataset
+        A collection of :term:`instances<Instance>`.
+
+    Feature
+        A pair of :term:`attribute` information and a corresponding value.
+
+    Feature vector
+        A vector that encodes information about single real object or an event.
+        Contains at least one :term:`feature`.
+
+    Inference
+        A process of obtaining outputs from the :term:`model` created on the :term:`training` stage
+
+    Instance
+        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
+        Each of :term:`labels<Label>` and :term:`features<Feature>` can encode different objects or events.
+
+    Label
+        :term:`Feature` of some object or event which is present on the :term:`training` stage but
+        unavaliable on the :term:`inference` stage.
+
+    Model
+        A result of :term:`training` machine-learning algorithm that stores information necessary to run
+        :term:`inference` on a new :term:`dataset`.
+
+    Observation
+        A synonym for :term:`instance`.
+
+    Prediction
+        - A synonym for :term:`inference`.
+        - An output of the :term:`trained<Training>` :term:`model` when provided with an input :term:`instance`.
+
+    Response
+        A value of a single :term:`label`. For example, response may be a flag
+        indicating that an observation belongs to a particular class.
+
+    Sample
+        A synonym for :term:`instance`.
+
+    Training
+        A search process of the best parameter values of the :term:`model`.
+
+    Variable
+        A synonym for an :term:`attribute`.
+
+|dal_short_name| library terms
+==============================
+
+.. glossary::
+    :sorted:
+
+    Batch Mode
+        The computation mode for an algorithm in |dal_short_name|, where all the data needed for
+        computation is available at the start and fits the memory of the device on which the computations are performed.
+
+    Immutability
+        The object is immutable when it is not possible to change its state after it is constructed.
+
+    Online Mode
+        The computation mode for an algorithm in |dal_short_name|, where the data needed for computation
+        becomes available in parts over time.
+
+    Workload
+        A task of applying a |dal_short_name| algorithm to a dataset.
+
+Other
+=====
+TODO: this section looks like a general one for the whole oneAPI spec.
```

Comment:
yes, it should be in the common glossary

Reply:
I mean, this section should be deleted when common glossary appears. Now I don't want to remove it because we have a references on the spec to these terms, which will be broken after removal. @rlnx @andreynv @outoftardis what is your opinion, should I delete it now?

Reply:
I'd have it here until a common mechanism for dictionaries/bibliographies is agreed and implemented on oneapi level

Reply:
I agree with @andreynv. But don't call this section "Other".  oneAPI terms, maybe? or common oneAPI terms?